packages/teleport/src/DesktopSession/TopBar.tsx

```diff
@@ -0,0 +1,167 @@
+  onDisconnect: VoidFunction;
+};
+
+const RecordingIndicator = styled.div``;
+
+export default function TopBar(props: TopBarProps) {
+  const { userHost, clipboard, recording, attempt, onDisconnect } = props;
+  const theme = useTheme();
+
+  const noneUntilSuccess = () => {
+    return {
+      display: attempt.status === 'success' ? 'flex' : 'none',
+    };
+  };
+
+  const showOnProcessing = () => {
+    return {
+      display: attempt.status === 'processing' ? 'flex' : 'none',
+    };
+  };
+
+  const showOnFailure = () => {
+    return {
+      display: attempt.status === 'failed' ? 'flex' : 'none',
+    };
+  };
+
+  const showOnDisconnected = () => {
+    return {
+      display: attempt.status === 'disconnected' ? 'flex' : 'none',
+    };
+  };
+
+  // Used for centering the middle component in the TopBar in certain states
+  const centeringDivStyle = () => {
+    return {
+      display:
+        attempt.status === 'processing' ||
+        attempt.status === 'disconnected' ||
+        attempt.status === 'failed'
+          ? 'flex'
+          : 'none',
+    };
+  };
+
+  const primaryOnTrue = (b: boolean): any => {
+    return {
+      color: b ? theme.colors.text.primary : theme.colors.text.secondary,
+    };
+  };
+
+  const userHostStyle = {
+    ...noneUntilSuccess(),
+    color: theme.colors.text.secondary,
+  };
+
+  const clipboardTextStyle = {
+    ...noneUntilSuccess(),
+    ...primaryOnTrue(clipboard),
+    verticalAlign: 'text-bottom',
+  };
+
+  const clipboardStyle = {
+    ...primaryOnTrue(clipboard),
+    fontWeight: theme.fontWeights.bold,
+    fontSize: theme.fontSizes[4] + 'px',
+    alignSelf: 'center',
+  };
+
+  const connectingStyle = {
+    ...showOnProcessing(),
+    color: theme.colors.text.secondary,
+    textAlign: 'center',
+  };
+
+  const errorStyle = {
+    ...showOnFailure(),
+    color: theme.colors.text.secondary,
+  };
+
+  const disconnectedStyle = {
+    ...showOnDisconnected(),
+    color: theme.colors.text.secondary,
+  };
+
+  const recordingTextStyle = primaryOnTrue(recording);
+
+  const recordingIndicatorStyle = {
+    width: '10px',
+    height: '10px',
+    borderRadius: '10px',
+    marginRight: '6px',
+    backgroundColor: recording
+      ? theme.colors.error.light
+      : theme.colors.text.secondary,
+    verticalAlign: 'text-bottom',
+  };
+
+  return (
+    <TopNav
+      height="56px"
+      bg={colors.dark}
+      style={{
+        justifyContent: 'space-between',
+      }}
+    >
+      <>
+        <div style={centeringDivStyle()}></div>
+        <Text px={3} style={userHostStyle}>
+          {userHost}
+        </Text>
+      </>
+
+      <>
+        {/* Center element. Only one of these is shown at a time depending on attempt.status */}
+        <Text style={clipboardTextStyle}>
+          <Clipboard style={clipboardStyle} pr={2} />
+          Clipboard Sharing {clipboard ? 'Enabled' : 'Disabled'}
+        </Text>
+        <Text style={connectingStyle}>Connecting...</Text>
```

Comment:
hm, wondering why we couldn't just inline the styles? 
```
{ attempt.status === 'processing' && <Text style={...}> }
```

or we could also pass the attempt down to a styled comp and do our checking for states there?
```
const SomeName = styled.div`
    color: ${({ attempt }) => attempt.state === 'proccessing' ? ....};
`;
```

Reply:
> hm, wondering why we couldn't just inline the styles?

We could. Initially I had it that way, but there was enough repetition that I elected for this approach. I was also finding it difficult to edit/read the JSX with so many ternary operator statements in there.

> or we could also pass the attempt down to a styled comp and do our checking for states there?

I had something like that initially but I prefer the `useTheme` hook. The styled comp has so much implicit "magic" in comparison.

Reply:
i think using styled components is the better approach but we can also get a second opinion from @alex-kovoy